spec/models/job_proxy_dispatcher_get_eligible_proxies_for_job_spec.rb

```diff
@@ -21,6 +21,9 @@
 
       context "with a vm scan job, " do
         before(:each) do
+          # admin user is needed to process Events
+          FactoryGirl.create(:user_with_group, :userid => "admin")
+          MiqEventDefinition.stub(:find_by_name => true)
```

Comment:
Due to the tenant support of automate, all events require the user/group/tenant information to find the correct automate domain for the event handling. When there is no user associated with an event, the admin user is used.

When an event for classes that support policy is raised, it is sent to automate only if the event is defined in MiqEventDefinition table. 
In testing, we can either create the event that is to be raised or stub MiqEventDefinition to treat every event as defined in the DB.